internal/consumer/consumer.go

```diff
@@ -332,14 +369,49 @@ func (i *InventoryConsumer) ProcessMessage(headers map[string]string, relationsE
 		i.Logger.Infof("processing message: operation=%s, txid=%s", operation, txid)
 		i.Logger.Debugf("processed message tuple=%s", msg.Value)
 		if relationsEnabled {
-			tuplesToReplicate, err := i.calculateTuplesOrNoOp(msg.Value, operation, biz.OperationTypeUpdated, txid)
+			// Parse the message into a TupleEvent
```

Comment:
The whole block appears to be similar to the one above:
 - Parse message
 - Check common version
 - Get current and previous
 - Calculate cuples
 - Skip if empty

Should we move this to a function and call this here and there?